### DISCUSSION TITLE
Question about RIS steering vector

### DISCUSSION THREAD

**Discussion by SStercek** [NVlabs/sionna#535] [Q&A]
Hello,

I am attempting to extract channel information (delay, AOD, etc.) and RIS phase information obtained from Sionna, and then reproduce this in MATLAB.

Assuming that both the TX and RX are single antennas, the RIS has a size of 32x32.

I positioned transmitters and receivers in the form of UPAs (Uniform Planar Arrays) with the same array structure as the RIS at the RIS location. Then, I acquired the channels between TX-RIS and RIS-RX, and created steering vectors (**h_tx, h_rx**) using the obtained angle information.
The phase information of the RIS was acquired using the` phase_gradient_reflector `to obtain the phase matrix **Phi,** so that the RIS could form a beam towards the defined RX location.
After that, I calculated the channel using the formula **h_rx^T * Phi * h_tx**  in MATLAB, but it seems that the expected result is not coming out.

Could you let me know if there is any problem with this approach?

Additionally, I derived the steering vectors by converting the AOD and AOA... information in the GCS to the LCS  (Note the [https://github.com/NVlabs/sionna/discussions/392](url) )and then taking the dot product with the positions of each element in the RIS. I am also curious if there is any issue with this part.

Thank you.

**Comment by jhoydis** [comment]
Hi @SStercek ,

In principle, your approach look ok to me. It is very difficult to guess where an error might occur.

Given that this setup is so simple that you can write down the result by hand, I would recommend that you check if the vectors $h_{rx}$ and $h_{tx}$ you obtain from Sionna match your expectations. 

Did you compute the phase matrix from elements $e^{j\phi_i}$, where $\phi_i$ is the phase for a specific RIS element?
Please also note that the phase gradient reflector does not beam toward a specific position, but rather a direction. The focusing lens does the former.

Did you use the flag `scene.synthetic_array=False`? If not, there might be a mismatch between the channels you compute and the once Sionna uses internally.

Hope that provides you with some ideas for debugging.
